applications/app/controllers/InteractiveController.scala

```diff
@@ -5,21 +5,35 @@ import common._
 import contentapi.ContentApiClient
 import conf._
 import conf.switches.Switches
+import model.Cached.RevalidatableResult
 import model._
+import play.api.libs.ws.WSClient
 import play.api.mvc._
 import views.support.RenderOtherStatus
+import conf.Configuration.interactive.cdnPath
+import conf.Configuration.environment.isPreview
 
 import scala.concurrent.Future
 
 case class InteractivePage (interactive: Interactive, related: RelatedContent) extends ContentPage {
   override lazy val item = interactive
 }
 
-class InteractiveController(contentApiClient: ContentApiClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
+class InteractiveController(contentApiClient: ContentApiClient, wsClient: WSClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
 
   def renderInteractiveJson(path: String): Action[AnyContent] = renderInteractive(path)
   def renderInteractive(path: String): Action[AnyContent] = Action.async { implicit request => renderItem(path) }
 
+  def renderInteractiveServiceworker(path: String): Action[AnyContent] = Action.async { implicit request =>
+    val stage = if (isPreview) "preview" else "live"
+    val serviceWorkerPath = s"$cdnPath/service-workers/$stage/$path/interactive-service-worker.js"
+    wsClient.url(serviceWorkerPath).get().map { response =>
+      Cached(60) {
```

Comment:
@Rich-Harris 

My point of view is more from an infrastructure perspective:
Ideally, the cache time of any assets on the guardian website is very high, with a cache busting mechanism when assets are modified. (ie: We use content hash to differentiate each version). File is then always served from the CDN and we avoid any round trip to our servers.

Having this service-worker endpoint outside of the normal assets deployment makes thing a little bit more complicated here. 
However the best solution would be to use a similar mechanism. I believe the interactive team uses timestamp, instead of hash, but it would be best imho to store the file at `/TIMESTAMP/server-worker-interactive.js` in your S3 bucket. This would allow us to cache it forever while being able to push a new version instantly and not worry about cache time from a developer perspective.

What do you think?